packages/ui/src/validators/modals/ClaimStakingRewardsModal/ClaimStakingRewardsModal.tsx

```diff
@@ -0,0 +1,445 @@
+      // This fills the block with maximum amount of eras
+      const calculatedMax = Math.floor((36 * 64) / maxNominatorRewarded)
+      return Math.max(1, calculatedMax) // Ensure at least 1
+    } catch {
+      return 5 // Fallback if calculation fails
+    }
+  }, [api])
+
+  const totals = useMemo(() => {
+    if (!validatorsRewards)
+      return {
+        totalClaimable: BN_ZERO,
+        totalEras: 0,
+        accountsCount: 0,
+        batchedEras: 0,
+        allPayouts: [] as Array<{ validatorAddress: string; era: number }>,
+      }
+
+    const totalClaimable = validatorsRewards.reduce((sum, v) => sum.add(v.totalClaimable), BN_ZERO)
+
+    // Create sorted list of all unclaimed payouts (validator + era pairs)
+    const allPayouts: Array<{ validatorAddress: string; era: number }> = []
+    validatorsRewards.forEach((validator) => {
+      validator.unclaimedEras.forEach((era) => {
+        if (!claimedEras.has(`${validator.address}-${era}`)) {
+          allPayouts.push({ validatorAddress: validator.address, era })
+        }
+      })
+    })
+
+    // Sort by era (ascending) to claim oldest first
+    allPayouts.sort((a, b) => a.era - b.era)
+
+    const batchedEras = Math.min(allPayouts.length, maxBatchSize)
+
+    return {
+      totalClaimable,
+      totalEras: allPayouts.length,
+      batchedEras,
+      accountsCount: validatorsRewards.length,
+      allPayouts, // Store sorted payouts for transaction creation
+    }
+  }, [validatorsRewards, claimedEras, maxBatchSize])
+
+  const transaction = useMemo(() => {
+    if (!api || !totals.allPayouts || totals.allPayouts.length === 0) return undefined
+
+    // Get the next batch of payouts (sorted by era, oldest first)
+    const nextBatch = totals.allPayouts.slice(0, maxBatchSize)
+
+    const payoutCalls = nextBatch.map(({ validatorAddress, era }) =>
+      api.tx.staking.payoutStakers(validatorAddress, era)
+    )
+
+    return payoutCalls.length === 0
+      ? undefined
+      : payoutCalls.length === 1
+      ? payoutCalls[0]
+      : api.tx.utility.batchAll(payoutCalls)
+  }, [api, totals.allPayouts, maxBatchSize])
+
+  const signerAccount = useMemo(() => {
+    if (!validatorsRewards || validatorsRewards.length === 0) {
+      if (activeMembership?.controllerAccount) {
+        return allAccounts.find((acc) => acc.address === activeMembership.controllerAccount) || allAccounts[0]
+      }
+      return allAccounts[0]
+    }
+    const firstController = validatorsRewards[0]?.controller
+    if (firstController) {
+      return allAccounts.find((acc) => acc.address === firstController) || allAccounts[0]
+    }
+    return allAccounts[0]
+  }, [activeMembership, allAccounts, validatorsRewards])
+
+  const { isReady, sign, paymentInfo, canAfford } = useSignAndSendTransaction({
+    transaction,
+    signer: signerAccount?.address ?? '',
+    service: service as any,
+    skipQueryNode: true,
+  })
+
+  // Automatically trigger next batch after transaction is finalized
```

Comment:
It shouldn't be necessary to wait for finalization to send new tx, just for the next block.